### PR TITLE
KOGITO-8345: [SWF Viewer] Icon frome metadata improvements

### PR DESCRIPTION
**Pull request by hasys**
Hi @handreyrc, @LuboTerifaj,

There is a fix for [KOGITO-8345](https://issues.redhat.com/browse/KOGITO-8345). Both issues are fixed:

* `metadata.icon` should overtake precedence from `metadata.type`
* if `metadata.icon` is incorrect, now always will be default state shown if `metadata.type` is not set or incorrect.

Since there are a lot of view code it's not so easy to write Unit tests, so I am working on it in separate PR as part of [KOGITO-8344](https://issues.redhat.com/browse/KOGITO-8344).

Thank you!

### CODE REVIEW

**Comment by hasys** [comment]
Hello @romartin, @LuboTerifaj,

I pushed related tests and those tests also covers [KOGITO-8344](https://issues.redhat.com/browse/KOGITO-8344).

PR is ready for review. 

Thank you!

**Comment by hasys** [comment]
Hi @romartin, @LuboTerifaj,

I fixed formatting issue, which failed previous build. Should be fine now. Thank you!

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/StateShape.java [hunk 1]
```diff
@@ -47,35 +48,25 @@ public StateShape(State state, ResourceContentService resourceContentService) {
...
         }
 
-        if (state.metadata.type != null) {
-            setType(state.metadata.type);
+        if (StringUtils.nonEmpty(state.metadata.icon)) {
+            if (state.metadata.icon.startsWith("data:")) {
+                Picture picture = new Picture(state.metadata.icon);
+                setIconPicture(picture);
+            } else {
+                loadIconFormFile(state, resourceContentService);
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/StateShape.java]:
```suggestion
                loadIconFromFile(state, resourceContentService);
```

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/StateShape.java [hunk 2]
```diff
@@ -85,6 +76,25 @@ public StateShape(String name) {
         shapeView = (StateShapeView) getShape().getShapeView();
     }
 
+    private void loadIconFormFile(State state, ResourceContentService resourceContentService) {
```
Comment (LuboTerifaj) [inline packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/StateShape.java]:
```suggestion
    private void loadIconFromFile(State state, ResourceContentService resourceContentService) {
```

**Review by LuboTerifaj — CHANGES_REQUESTED** [review]
Hello @hasys ,

The images are not scaled properly.

- I tested various resolutions.
- The circle backround colour in the diagram is taken from metadata type "ansible" (red), "kaoto" (purple), or blue for other values.
- The base64 image string scaling works well.

Thanks!

File: packages/serverless-workflow-diagram-editor/lienzo-core/src/main/java/com/ait/lienzo/client/core/util/Geometry.java [hunk 3]
```diff
@@ -653,11 +653,17 @@ public static final void drawArcJoinedLines(final PathPartList list, final PathP
 
             final PathPartEntryJSO nextEntry = baseList.get(i + 1);
 
-            Point2D p0 = basePoints.get(i - 1);
+            Point2D p0 = null;
```
Comment (romartin) [inline packages/serverless-workflow-diagram-editor/lienzo-core/src/main/java/com/ait/lienzo/client/core/util/Geometry.java]: in-lining the conditional initialization would help on reading this

File: packages/serverless-workflow-diagram-editor/lienzo-core/src/main/java/com/ait/lienzo/client/core/util/Geometry.java [hunk 4]
```diff
@@ -653,11 +653,17 @@ public static final void drawArcJoinedLines(final PathPartList list, final PathP
...
+            if (i > 0) {
+                p0 = basePoints.get(i - 1);
+            }
 
             final Point2D p2 = basePoints.get(i);
 
-            Point2D p4 = basePoints.get(i + 1);
+            Point2D p4 = null;
```
Comment (romartin) [inline packages/serverless-workflow-diagram-editor/lienzo-core/src/main/java/com/ait/lienzo/client/core/util/Geometry.java]: in-lining the conditional initialization would help on reading this

**Review by romartin — APPROVED** [review]
Looks good to me! Thanks @hasys

**Comment by hasys** [comment]
Thank you @romartin, @LuboTerifaj all notes fixed.

PR is ready again.

File: packages/serverless-workflow-diagram-editor/lienzo-core/src/main/java/com/ait/lienzo/client/core/util/Geometry.java (hunk 4 above)
Comment (hasys) [reply packages/serverless-workflow-diagram-editor/lienzo-core/src/main/java/com/ait/lienzo/client/core/util/Geometry.java]: fixed, thank you

File: packages/serverless-workflow-diagram-editor/lienzo-core/src/main/java/com/ait/lienzo/client/core/util/Geometry.java (hunk 3 above)
Comment (hasys) [reply packages/serverless-workflow-diagram-editor/lienzo-core/src/main/java/com/ait/lienzo/client/core/util/Geometry.java]: fixed, thank you

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/StateShape.java (hunk 2 above)
Comment (hasys) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/StateShape.java]: fixed, thank you

File: packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/StateShape.java (hunk 1 above)
Comment (hasys) [reply packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/StateShape.java]: fixed, thank you

**Review by LuboTerifaj — APPROVED** [review]
Hello @hasys ,

It works as expected now.

The only issue I found is present also on main. I reported it here:
https://issues.redhat.com/browse/KOGITO-8467

Thank you!

**Pull request merged**